contracts/contracts/L1/rollup/Rollup.sol

```diff
@@ -568,6 +570,26 @@ contract Rollup is OwnableUpgradeable, PausableUpgradeable, IRollup {
                 )
             );
 
+            require(_kzgData.length == 128, "invalid kzg data");
+
+            bytes memory _commitment = _kzgData[32:80];
+            bytes memory _xBytes = abi.encode(
+                keccak256(abi.encodePacked(_commitment, _publicInputHash))
+            );
+            // make sure x < BLS_MODULUS
+            _xBytes[0] = 0x0;
+
+            // [versioned_hash | x | y | commitment | proof]
+            // with x and y being padded 32 byte big endian values
+            bytes memory _input = abi.encode(blobhash(0), _xBytes, _kzgData);
+
+            bool ret;
+            bytes memory _output;
+            assembly {
+                ret := staticcall(gas(), 0x0a, _input, 0xc0, _output, 0x40)
+            }
+            require(ret, "prove failed");
+
             // verify batch
             IRollupVerifier(verifier).verifyAggregateProof(
```

Comment:
Use keccak256(abi.encodePacked(_publicInputHash, y)) as the public input for the aggregate proof.

Reply:
added